.github/workflows/nix_build_linux.yml

```diff
@@ -1,431 +1,348 @@
+          nix develop -c ninja -C build -k 0 rslang
+
+      - name: Build IR of the Rust examples
+        run: |
+          nix develop -c ninja -C build -k 0 compile_rust_examples
+          ls -al ./build/examples/rust/target/assigner-unknown-unknown/release/examples
+
+      - name: Build circuits(.crct) of the Rust examples
+        run: |
+          nix develop -c ninja -C build -k 0 rust_examples_generate_crct
+          ls -al ./build/examples/cpp
+
+      - name: Build assignment tables(.tbl) of the Rust examples
+        run: |
+          nix develop -c ninja -C build -k 0 rust_examples_generate_tbl
+          ls -al ./build/examples/cpp
+
+      - name: Build proof for the circuit of the Rust examples
+        run: |
+          nix develop -c ninja -C build -k 0 prove_rust_examples
+
+      - name: Build IR of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 compile_cpp_examples
+          ls -al ./examples/cpp
+
+      - name: Build circuits(.crct) of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 cpp_examples_generate_crct
+          ls -al ./examples/cpp
+
+      - name: Build assignment tables(.tbl) of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 cpp_examples_generate_tbl_no_check
+          ls -al ./examples/cpp
+
+      - name: Copy assigner output from separate generation mode
+        run: |
+          bash ./tests/copy_assigner_outputs.sh ./build/examples/cpp/ build/separate_generatuon
+
+      - name: Run simultaneous .tbl and .crct generation of the C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 cpp_examples_generate_both
+          ls -al ./examples/cpp
+
+      - name: Copy assigner output from simultaneous generation mode
+        run: |
+          bash ./tests/copy_assigner_outputs.sh ./build/examples/cpp/ build/simultaneous_generation
+
+      - name: Compare different assigner modes output
+        run: |
+          bash ./tests/compare_folders_content.sh build/separate_generatuon build/simultaneous_generation
+
+      - name: Run size estimation for C++ examples
+        run: |
+          nix develop -c ninja -C build -k 0 cpp_examples_estimate_size
+
+      - name: Copy examples' circuits and assignments for uploading
+        run: |
+          chmod +x ./examples/copy_artifacts.sh
+          ./examples/copy_artifacts.sh build/examples/cpp examples_output
+          ls -al examples_output/
+
+      - name: Set aritfacts' names
+        id: artifact-names
+        run: |
+          echo "transpiler=transpiler-output" >> $GITHUB_OUTPUT
+          echo "examples=circuits-and-assignments" >> $GITHUB_OUTPUT
+
+      - name: Upload examples' circuits and assignments artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.artifact-names.outputs.examples }}
+          path: |
+            examples_output

```

Comment:
We need this to be uploaded to nix cache for integration testing. As well as transpiler output

Reply:
will be done here
https://github.com/NilFoundation/zkLLVM/pull/623